application/helpers/expressions/em_manager_helper.php

```diff
@@ -8653,6 +8653,12 @@ public static function UpgradeQuestionAttributes($changeDB=false,$iSurveyID=NULL
         */
         private function getQuestionAttributesForEM($surveyid=0,$qid=0, $lang='')
         {
+            $cacheKey = $surveyid . $qid . $lang;
+            $value = Yii::app()->cache->get($cacheKey);
```

Comment:
Cache key should be prefixed with `em_cache` or something, so that we can clear all EM cache objects in one go.

Reply:
Why not create a emcache in internal ?
````
		'emcache'=>array(
			'class'=>!empty($userConfig['config']['disableemcache']) ? 'system.caching.CDummyCache' : 'CFileCache',
		),
````

Reply:
Wouldn't that lead to duplicated configuration? I'd assume cache and emcache would always be the same cache, no?

Reply:
current cache is used for DB caching only (and some Yii automatic cache (Language))

Yes, duplicate configuration but : it can be very interesting to have different cache system : one for big data, one for little data, one updated a lot, one not updated a lot etc …

For example : i think createFieldmap can be best in FileCache : because it's very huge (sometimes 2Mo or more …) and loaded one time by user.

Reply:
Else : Cache key should be prefixed  by EM + SURVEYID : we need to reset cache by survey, not globally (for all survey).